### PR TITLE
breakup org data load if too big or too slow

### PR DESCRIPTION
**Pull request by sarahentzel**

### CODE REVIEW

File: src/Repositories/OrgDataRepository.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        private bool CheckAdd(int check, object entity, DateTime dtBail,  int start, ref int completed, ref string data)
+        {
+            Debug.WriteLine(check.ToString() + ":" + DateTime.Now.ToString() + dtBail.ToString());
+            if (DateTime.Now > dtBail) return false;
+            if (start <= check)
+            {
+                string thisdata = jsonApiSerializer.Serialize(entity);
+                if (data.Length + thisdata.Length > (1000000 * 5.5))
+                    return false;
+                data += (check > start ?  "," : "") + thisdata;
```
Comment (gtryus) [inline src/Repositories/OrgDataRepository.cs]: so on line 45, we check for start <= check. Here we test for check > start?

File: src/Repositories/OrgDataRepository.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+        private async Task<IQueryable<OrgData>> GetData(IQueryable<OrgData> entities, string start)
+        {
+            string data = "{\"data\":[";
+            int iStart;
+            if (!int.TryParse(start, out iStart))
+                iStart = 0;
+            int iStartNext = iStart;
+            //give myself 20 seconds to get as much as I can...
+            DateTime dtBail = DateTime.Now.AddSeconds(10);
```
Comment (gtryus) [inline src/Repositories/OrgDataRepository.cs]: comment says 20 seconds. Code says 10 seconds.

File: src/Services/OrgDataService.cs [hunk 3]
```diff
@@ -4,6 +4,8 @@
 using SIL.Transcriber.Models;
 using SIL.Transcriber.Repositories;
 using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
```
Comment (gtryus) [inline src/Services/OrgDataService.cs]: do we use the compression?

File: src/Repositories/OrgDataRepository.cs (hunk 1 above)
Comment (sarahentzel) [reply src/Repositories/OrgDataRepository.cs]: perhaps it would be clearer if I changed it to check for the == case...

**Pull request merged**